k8s/client.go

```diff
@@ -448,3 +487,266 @@ func (client *Client) FindDeploymentByName(ctx context.Context, name string) (*a
+			Name:  "TOKEN",
+			Value: client.ProjectToken,
+		},
+	}
+	toCloud := apiv1.Container{
+		Name:            coreInstance.Name + "-sync-to-cloud",
+		Image:           toCloudImage,
+		ImagePullPolicy: apiv1.PullAlways,
+		Env:             env,
+	}
+	fromCloud := apiv1.Container{
+		Name:            coreInstance.Name + "-sync-from-cloud",
+		Image:           fromCloudImage,
+		ImagePullPolicy: apiv1.PullAlways,
+		Env:             env,
+	}
+
+	req := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Deployment",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      coreInstance.Name + "-sync",
+			Namespace: client.Namespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &deploymentReplicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: apiv1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: apiv1.PodSpec{
+					ServiceAccountName: serviceAccount,
+					Containers:         []apiv1.Container{fromCloud, toCloud},
+				},
+			},
+		},
+	}
+
+	options := metav1.CreateOptions{}
+
+	return client.AppsV1().Deployments(client.Namespace).Create(ctx, req, options)
+}
+
+func (client *Client) DeployOperator(ctx context.Context, version string) ([]ResourceRollBack, error) {
+	file, err := getOperatorManifest(version)
+	if err != nil {
+		return nil, err
+	}
+	applied, err := client.applyOperatorManifest(ctx, file)
+	if err != nil {
+		return nil, err
+	}
+	return applied, nil
+
+}
+
+func (client *Client) applyOperatorManifest(ctx context.Context, manifestFull []byte) ([]ResourceRollBack, error) {
+	dynamicClient, err := dynamic.NewForConfig(client.Config)
+	if err != nil {
+		return nil, err
+	}
+
+	manifests := splitManifest(manifestFull)
+
+	var appliedSuccessfully []ResourceRollBack
+	for _, manifest := range manifests {
+		manifest = strings.TrimSpace(manifest)
+		if manifest == "" {
+			continue
+		}
+
+		decoder := yamlk8s.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+		obj := &unstructured.Unstructured{}
+		_, gvk, err := decoder.Decode([]byte(manifest), nil, obj)
+		if err != nil {
+			return nil, err
+		}
+
+		//workaround to avoid creating a namespace, not needed if we remove the namespace from the manifest
+		if gvk.Kind == "Namespace" {
+			continue
+		}
+		namespace := obj.GetNamespace()
+		if namespace == defaultOperatorNamespace {
+			obj.SetNamespace(client.Namespace)
+		}
+
+		kindPluralized := strings.ToLower(gvk.Kind) + "s"
+		withResource := gvk.GroupVersion().WithResource(kindPluralized)
+		resource := dynamicClient.Resource(withResource)
+
+		//if already exists, skip
+		get, err := resource.Namespace(obj.GetNamespace()).Get(ctx, obj.GetName(), metav1.GetOptions{})
+		if err != nil {
+			if !k8serrors.IsNotFound(err) {
+				return nil, err
+			}
+		}
+		if get != nil {
+			continue
+		}
+
+		created, err := resource.Namespace(obj.GetNamespace()).Create(ctx, obj, metav1.CreateOptions{})
```

Comment:
Try something like this 
```
if withResource == "clusterrolebindings" {
			if crb, ok := obj.(rbacv1.ClusterRoleBindings); ok {
				crb.Subjects[0].Namespace = namespace
			}
		}

```

The cluster  role binding needs to refrence correct ServiceAccount (namespaced name) 
```
subjects:
  - kind: ServiceAccount
    name: controller-manager
-->    namespace: default
```